my_enumerable_module.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+# Extending Enumerable module with replicated methods for the following:
+# each, each_with_index, select, all?, none?, any?, count, map and inject
+#
+module Enumerable
+  def my_each
+    arr = self # this is to prevent a redundant-use-of-self error
+    0.upto(arr.length - 1) do |i|
+      yield(arr[i])
+    end
+  end
+
+  def my_each_with_index
+    arr = self # this is to prevent a redundant-use-of-self error
+    0.upto(arr.length - 1) do |i|
+      yield(arr[i], i)
+    end
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns an enumerator if no block is given

Reply:
corrected